### PR TITLE
Martin/5260 consensus matching metrics

### PR DESCRIPTION
**Pull request by arrivets**
This PR addresses issue [5260](https://github.com/dapperlabs/flow-go/issues/5260)

Added following metrics:

- `push_receipts_on_receipt_duration_seconds_total:` total seconds spent in `onReceipt` method, which is the function that processes incoming receipts from the `push-receipts` channel
- `on_approval_duration_seconds_total`: total seconds spent in `onApproval `method, which is the function that processes incoming approvals from the `push-approvals` AND `request-approvals-by-chunk`
- `check_sealing_duration_seconds_total`: total seconds spent in `checkingSealing` method

Note that onReceipt and onApproval call checkSealing **asynchronously** so the corresponding duration counters do not include the time spent in checkSealing

Added following spans:

```
con.matching.checkSealing
con.matching.checkSealing.sealableResults
con.matching.checkSealing.clearPools
con.matching.checkSealing.requestPendingReceipts
con.matching.checkSealing.requestPendingApprovals
con.matching.onReceipt
con.matching.onReceipt.validation
con.matching.onApproval
```

### CODE REVIEW

**Comment by AlexHentschel** [comment]
This PR adds multiple empty files `cmd/util/cmd/execution-state-extract/storage_report_<some number>.csv`. Are those required?

File: engine/consensus/matching/engine.go [hunk 1]
```diff
@@ -473,8 +496,7 @@ func (e *Engine) checkingSealing() error {
 	// skip if no results can be sealed yet
 	if len(sealableResults) == 0 {
 		return nil
-	}
-	if len(sealableResults) > 0 {
+	} else {
```
Comment (zhangchiqing) [inline engine/consensus/matching/engine.go:499]: the `else` seems unnecessary

**Review by zhangchiqing — APPROVED** [review]
Looks good. Just that there are a few csv files added, which I think could be removed.

File: module/metrics.go [hunk 2]
```diff
@@ -138,11 +138,17 @@ type ConsensusMetrics interface {
...
 	EmergencySeal()
+
+	// IncreaseOnReceiptDuration increases the number of seconds spent processing receipts
+	IncreaseOnReceiptDuration(duration time.Duration)
+
+	// IncreaseOnApprovalDuration increases the number of seconds spent processing approval
+	IncreaseOnApprovalDuration(duration time.Duration)
+
+	// IncreaseCheckSealingDuration records absolute time for the full sealing check by the consensus match engine
+	IncreaseCheckSealingDuration(duration time.Duration)
```
Comment (AlexHentschel) [inline module/metrics.go:151]: Currently, we implement the metric as a `Counter`. However, I feel this is an implementation detail internal to the metrics component. If we changed the metric type to Histogram, I feel the naming would be off. 

#### Suggestion:
How about we change the names to `OnReceiptProcessingDuration` `OnApprovalProcessingDuration` and `CheckSealingDuration` ?

**Review by AlexHentschel — APPROVED** [review]
looks great.

**Pull request merged**